logstash-core-event/lib/logstash/event.rb

```diff
@@ -98,7 +98,13 @@ def clone
       copy[k] = begin v.clone rescue v end
     end
 
-    self.class.new(copy)
+    cloned = self.class.new(copy)
+    
+    # Clone metadata too
+    @metadata.each do |k,v|
+      cloned["@metadata"][k] = begin v.clone rescue v end
+    end
+    cloned
```

Comment:
@wiibaa you are welcome anytime :)

I believe we should provide a `#clone` method to the user and this method should sanely does what is supposed to do.
Clone the object and make sure we drop any references, the Marshal dance in some cases is the only way to make it work. But has the side effect to be really slow in a tight event loop this is not something we want. (LogStash::Util.deep_clone is using marshal)

The problem is usually show if you do in place modification, mutability is a silly world :)

A quick search show 2 plugins using an explicit `Event#clone` the split filter and the dns filter.